juju/client/facade.py

```diff
@@ -296,6 +318,17 @@ def get_doc(self):
         return self._get_arg_str(True, "\n")
 
 
+def buildValidation(name, instance_type, instance_sub_type, ident=None):
+    INDENT = ident or "    "
```

Comment:
hmm, this is because we're generating the code vs doing the work? 

Normally I'd think that we'd have a function call and we'd pop calls onto a stack based on what we found instead of building if blocks, but I guess we're writing the code that does the validation vs actually doing it at all here right?

Reply:
Exactly, this is actually build a if for building a validation, rather than build a validation block. I would also rather have a stack based system, where we pass a list through. It makes it much easier to reason about. I might do that in another round of refactoring at a later date.